services/licenseStore.js

```diff
@@ -54,26 +86,22 @@ class LicenseStore {
       const template = licenseStrings.find(t => license.match(t));
       // eslint-disable-next-line no-continue
       if (typeof template === 'undefined') continue;
-      return this.selectLicense(license, template);
+      return selectLicense(license, template, this.portReferences);
     }
 
     return null;
   }
 
-  getLicense(id) {
-    return this.index[id];
+  // Returns the license with the passed id.
+  getLicenseById(id) {
+    return this.indices.id[id];
   }
 
-  // Returns the `license` with updated `url` from the list of portReferences
-  // if the `licenseString` is present as key in portReferences AND the `license`
-  // is in group 'ported'; otherwise it returns the original `license`.
-  selectLicense(license, licenseString) {
-    const url = this.portReferences[licenseString.toLowerCase()];
-
-    if (license.isInGroup('ported') && !!url) {
-      return buildPortedLicense(license, licenseString, url);
-    }
-    return license;
+  // Returns the first license with the passed name.
+  // Ordered by occurrence in `config/licenses/licenses.js`.
+  getLicenseByName(name) {
+    const id = this.indices.name[name][0];
```

Comment:
what if the license dies not exist? `this.indices.name["tessi"]` probably returns `undefined`. And as far as I know `undefined[0]` does not work in JS, right?